src/components/ListHeader.js

```diff
@@ -16,18 +16,35 @@ export const ListHeader = ({ data }) => {
     ).map((dataProvider, index) => ({
       id: index + 1,
       value: dataProvider,
-      selected: false
+      selected: dataProvider === queryVariables.dataProvider
     }));
 
   const [dropdownData, setDropdownData] = useState([
     {
       id: 0,
       value: '- Alle -',
-      selected: true
+      selected: !queryVariables.dataProvider
     },
     ...dataProviders
   ]);
 
+  const selectedDropdownData = dropdownData.find((entry) => entry.selected);
+
+  // https://medium.com/swlh/prevent-useeffects-callback-firing-during-initial-render-the-armchair-critic-f71bc0e03536
+  const initialRender = useRef(true);
+
+  // influence list data when changing selected dropdown value
+  // call update of the list with the selected data provider
+  useEffect(() => {
+    if (initialRender.current) {
+      initialRender.current = false;
+    } else {
+      // do not pass the value, if the id is 0, because we do not want to use "- Alle -" inside of
+      // updateListData
+      updateListData(!!selectedDropdownData.id && selectedDropdownData.value);
```

Comment:
how do we make sure that id > 0 ?

Reply:
`selectedDropdownData.id` is 0, 1, 2, 3, ...
checking the boolean interpretation of a integer results in the following:

```
!!0 => false
!!1 => true
!!2 => true
!!3 => true
```

so we can say `!!selectedDropdownData.id && selectedDropdownData.value` to ensure, that it is passing just values where id > 0